### ISSUE TITLE
Possible bug: `hasCodeWithoutSemanticChanges` not behaving as expected

### ISSUE THREAD

**Issue by adrianoapmartins** [reviewpad/reviewpad#805]
## Describe the bug

`hasCodeWithoutSemanticChanges` seems to be `true` to code that clearly has semantic changes.

This breaks:
- https://github.com/reviewpad/marketing/issues/85
- https://github.com/reviewpad/website/pull/192

## How to reproduce
Using the following `reviewpad.yml` configuration:
```yaml
Current config:

workflows:
  - name: automating-safe-approvals-for-low-risk-changes
    if:
      - rule: '!$isDraft()
            && $hasCodeWithoutSemanticChanges()
            && !$hasFilePattern("src/package.json")'
    then:
      - '$approve("LGTM: this PR contains only low-risk code changes")'
```
and suggest the following changes in a PR:
```diff
- var name = 'World';
- console.log(`Hello ${name}`);
+ var first = 'Hello';
+ var second = 'World';
+ var sentence = first + " " + second;
+ console.log(sentence);
```

Live examples:
https://github.com/adrianoapmartins/layout/pull/6
https://github.com/adrianoapmartins/layout/pull/7

## Expected behavior

`hasCodeWithoutSemanticChanges` returns false and fails the rule on the `reviewpad.yml` configuration

### Conversation
![Screenshot 2023-03-30 at 17 28 51](https://user-images.githubusercontent.com/38539/228903004-26c3682f-2833-4873-8aa9-5bf325d3ea3c.png)
### Files changed
![Screenshot 2023-03-30 at 17 29 05](https://user-images.githubusercontent.com/38539/228903051-c7e65792-b250-419d-98b5-88f6a522d3b4.png)
### Reviewpad Log
```log
2023-03-30T16:10:26Z [info] [START] request received
2023-03-30T16:10:26Z [info] Request details: app=gh_app_196990, event_type=pull_request, event_action=opened, delivery_id=5ec11b50-cf15-11ed-8637-d13f7f3270ce, repo_url=https://github.com/adrianoapmartins/layout
2023-03-30T16:10:27Z [info] Processing pull_request event
2023-03-30T16:10:27Z [info] Found pull request 7
2023-03-30T16:10:27Z [info] [REVIEWPAD-START] running reviewpad on pull request https://github.com/adrianoapmartins/layout/pull/7
2023-03-30T16:10:29Z [info] Analyzing workflow 'automating-safe-approvals-for-low-risk-changes'
2023-03-30T16:10:31Z [info] Evaluating workflow 'automating-safe-approvals-for-low-risk-changes'
2023-03-30T16:10:31Z [info] File code/hello-world.js is not ignored
2023-03-30T16:10:32Z [info] No semantic changes detected
2023-03-30T16:10:32Z [info] Rule '!$isDraft() && $hasCodeWithoutSemanticChanges() && !$hasFilePattern("src/package.json")' activated
2023-03-30T16:10:32Z [info] Executing program
2023-03-30T16:10:33Z [info] Creating review APPROVE with body LGTM: this PR contains only low-risk code changes
2023-03-30T16:10:34Z [info] Action $approve("LGTM: this PR contains only low-risk code changes") executed
2023-03-30T16:10:34Z [info] Execution done
2023-03-30T16:10:34Z [info] Generating report
```

**Comment by adrianoapmartins** [comment]
@marcelosousa  I need you to unblock this for me:
1. I'll check it out soon
2. Put this article on hold for an indefinite time, I'm not sure when I'll be able to check it.

The article: https://github.com/reviewpad/marketing/issues/85
